font-src/glyphs/letter/latin/upper-m.ptl

```diff
@@ -79,6 +80,23 @@ glyph-block Letter-Latin-Upper-M : begin
 				composite-proc sfT.lt.outer [NeedNotItalic sfT.rt.outer]
 					if para.isItalic sfB.lb.outer sfB.lb.full
 					if para.isItalic sfB.rb.outer sfB.rb.full
+	
+	define [CyrSoftEmShape] : params [top df form slabType slanted] : glyph-proc
+		local subDf : df.slice 4 3
+		local sidesSlope : if slanted 0.04 0
+		local xRightTop : subDf.rightSB - top * sidesSlope
+
+		local swSideBot : match form
+			[Just FORM-SAN-SMALL] : AdviceStroke 2.5  subDf.div
+			[Just FORM-SAN]       : AdviceStroke 2.5  subDf.div
+			[Just FORM-TURN-W]    : AdviceStroke 2.5  subDf.div
+			[Just FORM-HANGING]   : AdviceStroke 2.5  subDf.div
+			__                    : AdviceStroke 2.75 subDf.div
+
+		include : MShape top subDf form slabType slanted (kMidHang -- df.div)
```

Comment:
The `df.div` here uses a different DivFrame from the one passed into the function (which is sliced), so that the yMiddle height can be consistent with the regular Ms.